OutOfSchool/OutOfSchool.WebApi/Services/IOperationWithObjectService.cs

```diff
@@ -0,0 +1,51 @@
+﻿using OutOfSchool.Services.Enums;
+using OutOfSchool.WebApi.Models;
+using OutOfSchool.WebApi.Models.Notifications;
+using System.ComponentModel.DataAnnotations;
+
+namespace OutOfSchool.WebApi.Services;
+
+/// <summary>
+/// Defines interface for CRUD functionality for OperationWithObject entity.
+/// </summary>
+public interface IOperationWithObjectService
+{
+    /// <summary>
+    /// Add entity.
+    /// </summary>
+    /// <param name="operationType">Operation type to add.</param>
+    /// <param name="entityId">Id of entity to add.</param>
+    /// <param name="entityType">Entity type.</param>
+    /// <param name="eventDateTime">Date of the event.</param>
+    /// <param name="rowSeparator">Separator if we need it.</param>
+    /// <param name="comment">Comment.</param>
+    /// <returns>A <see cref="Task"/> representing the result of the asynchronous operation.</returns>
+    Task Create(
+        OperationWithObjectOperationType operationType,
+        Guid? entityId,
+        OperationWithObjectEntityType? entityType,
+        DateTimeOffset? eventDateTime = null,
+        string rowSeparator = null,
+        string comment = null);
+
+    /// <summary>
+    /// Delete the object from DB.
+    /// </summary>
+    /// <param name="id">Key of the OperationWithObject in table.</param>
+    /// <returns>A <see cref="Task"/> representing the result of the asynchronous operation.</returns>
+    Task Delete(Guid id);
+
+    /// <summary>
+    /// Get all operations with objects from the database.
+    /// </summary>
+    /// <param name="filter">Filter.</param>
+    /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
+    Task<IEnumerable<OperationWithObjectDto>> GetAll(OperationWithObjectFilter filter);
+
+    /// <summary>
+    /// Check if entity exists in DB.
+    /// </summary>
+    /// <param name="filter">Filter.</param>
+    /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
+    Task<bool> IsExists(OperationWithObjectFilter filter);
```

Comment:
`IsExists` does not sound good :) can change to just `Exists` or some other word with `Is` prefix :)